src/sig_stfl/sig_stfl.h

```diff
@@ -284,37 +283,46 @@ typedef struct OQS_SIG_STFL_SECRET_KEY {
+	/* Application-managed data related to secure storage of secret key data */
 	void *context;
 
 	/**
-	 * Secret Key retrieval Function
+	 * Serialize the stateful secret key.
 	 *
-	 * @param[in] sk The secret key represented as OQS_SIG_STFL_SECRET_KEY object
-	 * @param[out] sk_len length of the private key as a byte stream
-	 * @param[out] sk_buf_ptr pointer to private key data as a byte stream
-	 * @returns length of key material data available
-	 * Caller is responsible for **deallocating** the pointer to buffer `sk_buf_ptr`.
+	 * This function encodes the stateful secret key represented by `sk` into a byte stream
+	 * for storage or transfer. The `sk_buf_ptr` will point to the allocated memory containing
+	 * the byte stream. Users must free the `sk_buf_ptr` using `OQS_MEM_secure_free` after use.
+	 * The `sk_len` will contain the length of the byte stream.
+	 *
+	 * @param[out] sk_buf_ptr Pointer to the byte stream representing the serialized secret key.
+	 * @param[out] sk_len Pointer to the length of the serialized byte stream.
+	 * @param[in] sk Pointer to the `OQS_SIG_STFL_SECRET_KEY` object to serialize.
+	 * @return The number of bytes in the serialized byte stream upon success, or an OQS error code on failure.
+	 *
+	 * @remark The caller is responsible for ensuring that `sk` is a valid object before calling this function.
 	 */
 	OQS_STATUS (*serialize_key)(uint8_t **sk_buf_ptr, size_t *sk_len, const OQS_SIG_STFL_SECRET_KEY *sk);
 
 	/**
-	 * Secret Key to internal structure Function
+	 * Deserialize a byte stream into the internal representation of a stateful secret key.
+	 *
+	 * This function takes a series of bytes representing a stateful secret key and initializes
+	 * the internal `OQS_SIG_STFL_SECRET_KEY` object with the key material. This is particularly
+	 * useful for reconstructing key objects from persisted or transmitted state.
 	 *
-	 * @param[in] sk OQS_SIG_STFL_SECRET_KEY object
-	 * @param[in] key_len length of the returned byte string
-	 * @param[in] sk_buf The secret key data to populate the key object
-	 * @param[in] context application-specific data
-	 *            used to keep track of this secret key stored in a secure manner.
-	 *            The application manages this memory.
-	 * @returns  status of the operation populated with key material none zero length.
-	 * Caller is responsible to **unallocate** the buffer `sk_buf`.
+	 * @param[out] sk Pointer to an uninitialized `OQS_SIG_STFL_SECRET_KEY` object to hold the secret key.
+	 * @param[in] sk_len The length of the secret key byte stream.
+	 * @param[in] sk_buf Pointer to the byte stream containing the serialized secret key data.
+	 * @param[in] context Pointer to application-specific data, handled externally, associated with the key.
+	 * @returns OQS_SUCCESS if the deserialization succeeds, with the `sk` object populated with the key material.
+	 *
+	 * @remark The caller must ensure that resources (e.g., memory for `sk_buf`) are properly managed.
```

Comment:
What is the meaning of "properly managed"? Would it be better to state that explicitly (who/when alloc/free need to be called; is both heap and stack allocation OK?)?

Reply:
I updated the doc, but I'm not sure it's right. 
If the `sk_buf` is stack, then user don't have to deallocate. So I say a generic one "properly managed". 
I changed to 
`* @attention The caller is responsible for ensuring that `sk_buf` is securely deallocated when it's no longer needed.` in https://github.com/open-quantum-safe/liboqs/pull/1655/commits/c577c2ed8d6597c1766f5d9612397ed876061aa3